iothub/device/src/Transport/Amqp/AmqpUnit.cs

```diff
@@ -528,16 +528,22 @@ private async Task OpenMethodsReceiverLinkAsync(AmqpIoTSession amqpIoTSession, s
             }
         }
 
-        public async Task DisableTwinPatchAsync(TimeSpan timeout)
+        public async Task DisableTwinLinksAsync(TimeSpan timeout)
         {
             if (Logging.IsEnabled)
             {
-                Logging.Enter(this, timeout, $"{nameof(DisableTwinPatchAsync)}");
+                Logging.Enter(this, timeout, $"{nameof(DisableTwinLinksAsync)}");
             }
 
             Debug.Assert(_twinSendingLink != null);
             Debug.Assert(_twinReceivingLink != null);
 
+            bool gain = await _twinLinksSemaphore.WaitAsync(timeout).ConfigureAwait(false);
```

Comment:
what is the timeout here?

Reply:
It's operation timeout same as what is used with _messageReceivingLinkSemaphore